File: tests/fixtures/golden/typescript/animals.ts
export interface Speaker {
  say(): string;
}

export class Animal {
  // Base method `Dog` inherits without overriding.
  speak(): string {
    return "...";
  }
}

export class Dog extends Animal {
  bark(): string {
    return "Woof";
  }
}

// Decoy: an unrelated class with its own `speak` and `say`.
export class Cat implements Speaker {
  speak(): string {
    return "Meow";
  }

  say(): string {
    return "Meow";
  }
}

File: tests/fixtures/golden/csharp/Animals.cs
namespace Golden.Animals
{
    public interface ISpeaker
    {
        string Say();
    }

    public class Animal
    {
        // Base method `Dog` inherits without overriding.
        public string Speak() => "...";
    }

    public class Dog : Animal
    {
        public string Bark() => "Woof";
    }

    // Decoy: an unrelated class with its own `Speak` and `Say`.
    public class Cat : ISpeaker
    {
        public string Speak() => "Meow";

        public string Say() => "Meow";
    }
}
